packages/client/hmi-client/src/views/Model.vue

```diff
@@ -0,0 +1,143 @@
+	I: [
+		{ it: 1, is: 1 },
+		{ it: 1, is: 2 },
+		{ it: 2, is: 2 },
+		{ it: 3, is: 2 }
+	],
+	O: [
+		{ ot: 1, os: 2 },
+		{ ot: 1, os: 2 },
+		{ ot: 2, os: 3 },
+		{ ot: 3, os: 4 }
+	]
+};
+class ModelPlanRenderer extends graphScaffolder.BasicRenderer<NodeData, EdgeData> {
+	setupDefs() {
+		const svg = d3.select(this.svgEl);
+
+		// Clean up
+		svg.select('defs').selectAll('.edge-marker-end').remove();
+
+		// Arrow defs
+		svg
+			.select('defs')
+			.append('marker')
+			.classed('edge-marker-end', true)
+			.attr('id', 'arrowhead')
+			.attr('viewBox', MARKER_VIEWBOX)
+			.attr('refX', 2)
+			.attr('refY', 0)
+			.attr('orient', 'auto')
+			.attr('markerWidth', 15)
+			.attr('markerHeight', 15)
+			.attr('markerUnits', 'userSpaceOnUse')
+			.attr('xoverflow', 'visible')
+			.append('svg:path')
+			.attr('d', ARROW)
+			.style('fill', '#000')
+			.style('stroke', 'none');
+	}
+
+	renderNodes(selection: D3SelectionINode<NodeData>) {
+		const state = selection.filter((d) => d.data.type === NodeType.State);
+		const transitions = selection.filter((d) => d.data.type === NodeType.Transition);
+
+		transitions
+			.append('rect')
+			.classed('shape', true)
+			.attr('width', (d) => d.width)
+			.attr('height', (d) => d.height)
+			.style('fill', '#88C')
+			.style('stroke', '#888');
+
+		state
+			.append('circle')
+			.classed('shape', true)
+			.attr('cx', (d) => d.width * 0.5)
+			.attr('cy', (d) => d.height * 0.5)
+			.attr('r', (d) => d.width * 0.5)
+			.attr('fill', '#f80');
+
+		selection
+			.append('text')
+			.attr('y', -5)
+			.text((d) => d.label);
+	}
+
+	renderEdges(selection: D3SelectionIEdge<EdgeData>) {
+		selection
+			.append('path')
+			.attr('d', (d) => pathFn(d.points))
+			.style('fill', 'none')
+			.style('stroke', '#000')
+			.style('stroke-width', 2)
+			.attr('marker-end', 'url(#arrowhead)');
+	}
+}
+
+onMounted(async () => {
+	let renderer: ModelPlanRenderer | null = null;
+	const modelDrawnElement = document.getElementById('model-panel') as HTMLDivElement;
+	const g: IGraph<NodeData, EdgeData> = parsePetriNet2IGraph(model); // get graph from petri net representation
+
+	renderer = new ModelPlanRenderer({
+		el: modelDrawnElement,
+		useAStarRouting: true,
+		runLayout: runDagreLayout
+	});
+
+	// Test on click
+	renderer.on(
+		'node-click',
+		(_eventName: string | symbol, _event, selection: D3SelectionINode<NodeData>) => {
+			console.log(selection.datum());
+		}
+	);
+
+	// write json to model-json and draw model to model-drawn:
+	d3.select('#model-json').text(JSON.stringify(model));
+	await renderer?.setData(g);
+	await renderer?.render();
+});
+</script>
+
+<template>
+	<section class="model">
+		<p>Model Renderer</p>
+		<div style="display: flex">
+			<div id="model-panel" class="model-panel"></div>
+			<div id="model-json" class="model-panel"></div>
+		</div>
+	</section>
+</template>
+
+<style scoped>
+.model {
+	margin: 10px;
+}
+
+.model-panel {
+	width: 500px;
+	height: 500px;
+	border: 1px solid #888;
```

Comment:
Please use Uncharted Design Tokens black colours
https://github.com/unchartedsoftware/design-tokens/blob/main/build/css/uncharted-tokens.css#L11-L22

```suggestion
	border: 1px solid var(--un-color-black-40);
```